sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -601,14 +601,24 @@ aggregationClause
     ;
 
 groupByClause
-    : groupingAnalytics
+    : nestedGroupingSets
+    | groupingAnalytics
     | expression
     ;
 
+nestedGroupingSets
+    :  GROUPING SETS '(' nestedGroupingSet (',' nestedGroupingSet)* ')'
```

Comment:
What if we have (2 or more)-nested grouping sets? `GROUPING SETS(GROUPING SETS(GROUPING SETS(...`? I think we need tests for it.

Reply:
> What if we have (2 or more)-nested grouping sets? `GROUPING SETS(GROUPING SETS(GROUPING SETS(...`? I think we need tests for it.

Good catch..Just tested, PG support nested grouping sets.